src/libs/meta/meta.c

```diff
@@ -1211,7 +1211,7 @@ int elektraSortTopology (KeySet * ks, Key ** array)
 					retVal = -1;
 					break;
 				}
-				else if (i == j)
+				else if ((long)i == (long)j)
 				{
```

Comment:
Btw. is it difficult to ignore missing deps?

I am thinking about an optimization where I could add new keys on demand, thus deps that did not exist before might exist for the next sort.

From usability point you could check if deps are correct keynames (try if a keyName() has the same content as what you gave with keySetName()) and return -1 on the cases where there is such a problem.